test/Autofac.Test.CodeGen/Helpers/ModuleInitializer.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Autofac Project. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Runtime.CompilerServices;
+using Microsoft.CodeAnalysis;
+
+namespace Autofac.Test.CodeGen.Helpers;
+
+public static class ModuleInitializer
+{
+    [ModuleInitializer]
+    public static void Init()
+    {
+        // Override the output converter so we can filter out files we're not interested in.
+        VerifierSettings.RegisterFileConverter<GeneratorDriverRunResult>(ConvertRunResult);
+        VerifierSettings.RegisterFileConverter<GeneratorDriver>(ConvertDriver);
+
+        VerifySourceGenerators.Enable();
+    }
+
+    private static ConversionResult ConvertRunResult(GeneratorDriverRunResult target, IReadOnlyDictionary<string, object> context)
+    {
+        var exceptions = new List<Exception>();
+        var targets = new List<Target>();
+        foreach (var result in target.Results)
+        {
+            if (result.Exception != null)
+            {
+                exceptions.Add(result.Exception);
+            }
+
+            targets.AddRange(result.GeneratedSources.Where(x => x.HintName.EndsWith(".g.cs")).Select(SourceToTarget));
+        }
+
+        if (exceptions.Count == 1)
+        {
+            throw exceptions.First();
+        }
+
+        if (exceptions.Count > 1)
+        {
+            throw new AggregateException(exceptions);
+        }
+
+        if (target.Diagnostics.Any())
+        {
+            var info = new
+            {
+                target.Diagnostics,
+            };
+            return new(info, targets);
+        }
+
+        return new(null, targets);
+    }
+
+    private static ConversionResult ConvertDriver(GeneratorDriver target, IReadOnlyDictionary<string, object> context)
+    {
+        return ConvertRunResult(target.GetRunResult(), context);
+    }
+
+    private static Target SourceToTarget(GeneratedSourceResult source)
+    {
+        var data = $@"//HintName: {source.HintName}
```

Comment:
```suggestion
        var data = $@"// HintName: {source.HintName}
```
Not sure if we have control over this, but the analyzers would like it better if there was a space.

Reply:
I think a space may not be allowed, but will check. Why does the analyzer not understand the comment is a string...?

Reply:
Analyzers want a space between `//` and the comment text. Admittedly it's not that big of a deal since it seems mostly to support the `Verify` framework and it doesn't show up (at least in VS Code) unless you actually open that file since it's excluded from compilation. It was just nice to be able to go down the list of analyzers and go, "Yup, the generated code adheres here, doesn't here... OK, we know how to tweak it." If we can't change this, it's not going to break my heart. 💔

Reply:
Looks like a space is fine; so you can rest easy.